AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/Support/MutationEventExtensionsTests.swift

```diff
@@ -187,16 +227,320 @@ class MutationEventExtensionsTest: BaseDataStoreTests {
+            }
+        }
+        wait(for: [queryAfterUpdatingVersionExpectation], timeout: 1)
+    }
+
+    /// - Given: A pending mutation events queue with event containing version 2, a sent mutation event model that matches the
+    ///         received mutation sync model having version 2. The received mutation sync has version 1.
+    /// - When: The sent model matches the received model and the first pending mutation event version is 2.
+    /// - Then: The first pending mutation event version should NOT be updated.
+    func testSentModelVersionNewerThanResponseVersion_PendingEventNotReconciled() throws {
+        let modelId = UUID().uuidString
+        let post1 = Post(id: modelId, title: "title1", content: "content1", createdAt: .now())
+        let post2 = Post(id: modelId, title: "title2", content: "content2", createdAt: .now())
+        let requestMutationEvent = MutationEvent(id: UUID().uuidString,
+                                                modelId: modelId,
+                                                modelName: Post.modelName,
+                                                json: try post1.toJSON(),
+                                                mutationType: .create,
+                                                version: 2,
+                                                inProcess: true)
+
+        let pendingMutationEvent = MutationEvent(id: UUID().uuidString,
+                                                        modelId: modelId,
+                                                        modelName: Post.modelName,
+                                                        json: try post2.toJSON(),
+                                                        mutationType: .update,
+                                                        version: 2)
+
+        let metadata = MutationSyncMetadata(id: modelId,
+                                            deleted: false,
+                                            lastChangedAt: Int(Date().timeIntervalSince1970),
+                                            version: 1)
+
+        let responseMutationSync = MutationSync(model: AnyModel(post1), syncMetadata: metadata)
+
+        let requestMutationEventExpectation = expectation(description: "save requestMutationEvent success")
+        storageAdapter.save(requestMutationEvent) { result in
+            guard case .success = result else {
+                XCTFail("Failed to save metadata")
+                return
+            }
+            requestMutationEventExpectation.fulfill()
+        }
+
+        let pendingMutationEventExpectation = expectation(description: "save pendingMutationEvent success")
+        storageAdapter.save(pendingMutationEvent) { result in
+            guard case .success = result else {
+                XCTFail("Failed to save metadata")
+                return
+            }
+            pendingMutationEventExpectation.fulfill()
+        }
```

Comment:
this can be reduced to a set up step that invovles saving an array of pending mutation events, and verifying that the head is as expected:
```
setUpPendingMutationQueue([requestMutationEvent, pendingMutationEvent])
```


```
func setUpPendingMutationQueue(_ mutationEvents: [MutationEvent]) {
   // for each mutationEvent, perform storageAdapter.save

  // verify the head of the queue is as expected
}
```